src/figuresClasses.ts

```diff
@@ -1,11 +1,105 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+type Color = 'red' | 'green' | 'blue';
 
-export class Circle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: Color, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Any length must not be less than 0');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error(
+        'The longest side of a triangle is greater than a sum of two others',
+      );
+    }
+
+    this.color = color;
+
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
```

Comment:
u can use public to declare and initialize ur variables in the constructor. This will make ur code smaller
```suggestion
  shape: Shape = 'triangle';

  constructor(
    public color: Color,
    public a: number, 
    public b: number,
    public c: number
  ) {
    if (a <= 0 || b <= 0 || c <= 0) {
      throw new Error('Any length must not be less than 0');
    }

    if (a + b <= c || a + c <= b || b + c <= a) {
      throw new Error(
        'The longest side of a triangle is greater than a sum of two others',
      );
    }
  }
```